src/util/expr.cpp

```diff
@@ -245,7 +245,7 @@ bool exprt::is_zero() const
     {
       rationalt rat_value;
       if(to_rational(*this, rat_value))
-        assert(false);
+        CHECK_RETURN(false);
```

Comment:
Yes...  this patch set is intended to replace assert but there are a few other invariant handling patterns that could benefit from some kind of support.  This is one of them, another is:

```c
if (invariant) {
  debug() << "Actually helpful message with " << variables << " in it "
                << "maybe even " << e.pretty();
  assert(0); 
}
```

We should do something about these but I don't think we should do so in this patch set.  I this will come out of trying to convert things to this way of doing things.  We will get a better idea of what patterns there are and how they could be improved.